tunnel-server/src/proxy.ts

```diff
@@ -52,6 +58,24 @@ export function proxyHandlers({
         return;
       }
 
+      if(env.access === 'private'){
+        const authorization = req.headers['authorization'];
+        if (!authorization) {
+          return unauthorized(res)
+        }
+
+        const basicAuth = Buffer.from(authorization?.split('Basic ')[1] || '', 'base64').toString('ascii');
```

Comment:
Might worth splitting the authentication scheme and check if it's basic.
http://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml